### ISSUE TITLE
Connecting to Fanuc controller

### ISSUE THREAD

**Issue by mmaniscalco** [digitalpetri/ethernet-ip#38]
I'm using your package to write a client that can connect directly to a robotic controller instead of going through PLC.  Everything seems to work well until I try to use connection pools.  

Connection pools are throwing the following exception:
java.util.concurrent.ExecutionException: com.digitalpetri.enip.cip.CipResponseException: status=0x01 [connection failure] , additional=[0x0315]

Extended status error 0x0315 says, "This error occurs when the target device cannot understand the segment type or segment value in the Connection Path. Consult the documentation of the device to verify the correct segment type and value. If a Connection Point greater than 255 is specified this error could occur."  

I am using the same connection path in my connection pool version and non-connection pool version.  I did notice that there is another connection path in the call to ForwardOpenRequest.  Is this the one it is complaining about?

Can I even use connections pool when connecting directly to a robotic controller.

**Comment by kevinherron** [comment]
I haven’t connected to one of these so I’m not sure what its limitations might be.

It could be that it only supports “Unconnected” messaging, and so when you try to use “Connected” messaging with the pool you get this error, or it could be that you’ve supplied the wrong connection path. The correct connection path would come from either documentation or possibly the EDS file.

**Comment by mmaniscalco** [comment]
Thanks for the quick reply.

We did figure it out.  The issue was that we were setting the connection path like follows.

EPath.PaddedEPath connectionPath = new EPath.PaddedEPath(new PortSegment(1, new byte[]{(byte) 0}));

and giving this to the client 

CipClient client = new CipClient(config, connectionPath);

This was interfering with the connection path that was being appended in the CipClient class for MESSAGE_ROUTER_CP_PATH.  Once we changed the above connection path to

private static final EPath.PaddedEPath connectionPath = new EPath.PaddedEPath(Collections.emptyList());

everything started working.

Thanks!
